test/test.js

```diff
@@ -51,18 +53,19 @@ describe('request-progress', function () {
         });
 
         request.emit('request');
-        request.emit('response', { headers: { 'content-length': 10 } });
+        response.headers = { 'content-length': 10 };
+        request.emit('response', response);
```

Comment:
I would rather use:

```javascript
request.emit('response', Object.assign(response, { headers: { 'content-length': 10 } });
```

But this could ruin support for older nodes, so I went with the simplest approach

Opinions?